processing/src/main/java/org/apache/druid/segment/UnnestDimensionCursor.java

```diff
@@ -409,7 +412,12 @@ public int size()
     @Override
     public int get(int idx)
     {
-      return indexedIntsForCurrentRow.get(index);
+      // to support rows which have only null values
+      // need to check if the value is not null and the size is greater than 0
+      if (indexedIntsForCurrentRow != null && indexedIntsForCurrentRow.size() > 0) {
+        return indexedIntsForCurrentRow.get(index);
+      }
```

Comment:
Why would we be getting any rows at all when unnesting a data point that was null to begin with?  That is, if we had the array-of-null, we should be getting an IndexedInts back with 1 entry, which is null.  If we didn't have anything to begin with, then we should be getting back `null` and probably don't have any work to do on the row anyway?

Reply:
The comment is misleading, I'll update the comment we process the row only if it is not null and has atleast 1 value